tf2_ros_py/tf2_ros/buffer_client.py

```diff
@@ -79,7 +80,8 @@ def __init__(
         """
         tf2_ros.BufferInterface.__init__(self)
         self.node = node
-        self.action_client = ActionClient(node, LookupTransform, action_name=ns)
+        # self.action_client = ActionClient(node, LookupTransform, action_name=ns)
+        self.service_client = node.create_client(LookupTransform, ns, callback_group=MutuallyExclusiveCallbackGroup())
```

Comment:
In a lot of cases, the nodes containing a buffer client will need a MultiThreadExecutor to avoid deadlocks. one common case is if the buffer client call is in a ROS Timer